readability/readability.py

```diff
@@ -155,7 +157,10 @@ def _html(self, force=False):
         return self.html
 
     def _parse(self, input):
-        doc, self.encoding = build_doc(input)
+        if isinstance(input, (_ElementTree, HtmlElement)):
+            doc = input
```

Comment:
This option doesn't set self.encoding which was set in another branch.
Users might be relying on it.

Reply:
Also, you have no deepcopy as we have discussed.
Any thoughts on this?
._parse method could be run several times, devastating to the document after the first iteration and thus reducing the found subtree quality.

Reply:
I added the encoding attribute, for the `deepcopy` issue I'm not sure what to do, passing already existing trees is experimental anyway